### ISSUE TITLE
buildbot.test.integration.test_worker_workerside.TestWorkerConnection.test_pb_keepalive is flaky

### ISSUE THREAD

**Issue by p12tic** [buildbot/buildbot#4969]
Examples: https://buildbot.buildbot.net/#/builders/8/builds/1856/steps/8/logs/stdio,  https://buildbot.buildbot.net/#/builders/23/builds/1149/steps/8/logs/stdio. Note the different failure reasons.

**Comment by rajdeepbharati** [comment]
This passed on my local setup

**Comment by p12tic** [comment]
The point is that the test fails only on some tests, only in certain conditions which are for some reason more prevalent on the CI environment.

**Comment by p12tic** [comment]
In the end I was able to reproduce various issues with the test by running many instances of it in parallel while limiting the total cpu usage to a single core so that the tests are preempted many times and hopefully interleave the deferreds differently. Looks like the keepalive may be called more than once for some reason.

**Comment by rajdeepbharati** [comment]
Ok, pretty cool!

**Comment by rajdeepbharati** [comment]
Ok, pretty cool!

**Comment by p12tic** [comment]
Still happens.

**Comment by p12tic** [comment]
I'll open another issue for this bug.
